Algebra.java

```diff
@@ -7,61 +7,251 @@ public class Algebra {
+		System.out.println(plus(4,plus(4,4)));
+		System.out.println("minus");
+		System.out.println(plus(-4,plus(-4,-4)));
 	    System.out.println(minus(7,2));  // 7 - 2
    		System.out.println(minus(2,7));  // 2 - 7
+		System.out.println(minus(-2,7));  // -2 - 7
+		System.out.println(minus(-2,-7));  // -2 -(-7)
+		System.out.println(minus(-7,-2));
+		System.out.println("times");
  		System.out.println(times(3,4));  // 3 * 4
+		System.out.println(times(-3,4));  // -3 * 4
+		System.out.println(times(1,-1));
    		System.out.println(plus(2,times(4,2)));  // 2 + 4 * 2
+		System.out.println("power");
    		System.out.println(pow(5,3));      // 5^3
    		System.out.println(pow(3,5));      // 3^5
+		System.out.println(pow(-3,5));
+		System.out.println("divider");
    		System.out.println(div(12,3));   // 12 / 3    
    		System.out.println(div(5,5));    // 5 / 5  
    		System.out.println(div(25,7));   // 25 / 7
+		System.out.println(div(7,25));   // 7 / 25
+		System.out.println(div(7,4));   // 7 / 4    
+   		System.out.println(div(7,-4));    // 7 / -4  
+   		System.out.println(div(-7,-4));   // -7 / -4
+		System.out.println(div(-7,4));   // -7 / 4
+		System.out.println(div(-4,7));
+		System.out.println(div(-4,-7));
+		System.out.println(div(4,-7));
+		System.out.println("mod");
    		System.out.println(mod(25,7));   // 25 % 7
    		System.out.println(mod(120,6));  // 120 % 6    
+		System.out.println(mod(7,-4));
+		System.out.println(7%-4);
+		System.out.println(mod(-7,-4));
+		System.out.println(-7%-4);
+		System.out.println(mod(-7,4));
+		System.out.println(-7%4);
+		System.out.println(mod(4,7));
+		System.out.println(4%7);
+		System.out.println(mod(4,-7));
+		System.out.println(4%-7);
+		System.out.println(mod(-4,-7));
+		System.out.println(-4%-7);
+		System.out.println("sqrt");
    		System.out.println(sqrt(36));
 		System.out.println(sqrt(263169));
    		System.out.println(sqrt(76123));
+		System.out.println(sqrt(7));
 	}  
 
+
 	// Returns x1 + x2
 	public static int plus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		while (x2 != 0){
+
+			if(x2 == 0){
+				return x1;
+			}
+
+			if (x2 > 0){
+				x1++;
+				x2--;
+
+			}
+			else{
+				x1--;
+				x2++;
+			}
+		}
+		return x1;
 	}
 
+
 	// Returns x1 - x2
 	public static int minus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		if (x2 == 0){
+			return x1;
+		}
+		while (x2!=0) { 
+			if(x2 > 0){
+				x1--;
+				x2--;
+			}
+			else{
+				x1++;
+				x2++;
+			}
+			
+		}
+		return x1;
 	}
 
-	// Returns x1 * x2
+
 	public static int times(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int original = x1;
+		if (x2 == 0 || x1 == 0){
+			return 0;
+		}
+		else if (x2 == 1){
+			return x1;
+		}
+		else {
+			if (x1 > 0 && x2 >0){
+				x2--;
+				while(x2 != 0){
+					x1 = plus (x1, original);
+					x2--;
+				}
+			}
+			//maybe i can put together
+			else if (x1 < 0 && x2 > 0) {
```

Comment:
It is not wrong to check all possible combinations of `x1` and `x2`, but there is a more clean, short, elegant solution.
It is also better for you as a programmer to think of shorter implementation (so you won't have to write a lot...)